sdk/extensions/Microsoft.Extensions.Azure/src/Internal/ClientFactory.cs

```diff
@@ -93,6 +94,15 @@ internal static TokenCredential CreateCredential(IConfiguration configuration, T
             var certificate = configuration["clientCertificate"];
             var certificateStoreName = configuration["clientCertificateStoreName"];
             var certificateStoreLocation = configuration["clientCertificateStoreLocation"];
+            var additionallyAllowedTenants = configuration["additionallyAllowedTenants"];
+            IEnumerable<string> additionallyAllowedTenantsList = null;
+            if (!string.IsNullOrWhiteSpace(additionallyAllowedTenants))
+            {
+                // not relying on StringSplitOptions.RemoveEmptyEntries as we want to remove leading/trailing whitespace between entries
+                additionallyAllowedTenantsList = additionallyAllowedTenants.Split(TenantDelimiter)
+                    .Select(t => t.Trim())
```

Comment:
I believe this will throw if `t` is `null`.   Should we use `t?.Trim()` instead?

Reply:
I don't see how `t` can be `null` here. Do you have a scenario?

Reply:
No, I don't.   I was thinking of the case where there was a double-delimiter, but that comes back as empty.   In that case, the `where` can be simplified to `Where(t => t != string.Empty)`, I think.